app/services/make_offer.rb

```diff
@@ -12,17 +14,35 @@ def initialize(actor:,
   end
 
   def save!
-    make_an_offer = MakeAnOffer.new(actor: actor,
-                                    application_choice: application_choice,
-                                    course_option: course_option,
-                                    offer_conditions: conditions)
-
-    unless make_an_offer.save
-      if make_an_offer.errors[:base].include?(MakeAnOffer::STATE_TRANSITION_ERROR)
-        ApplicationStateChange.new(application_choice).make_offer!
-      else
-        raise 'Unable to complete save on make_an_offer'
+    auth.assert_can_make_decisions!(application_choice: application_choice, course_option: course_option)
+
+    if offer.valid?
+      audit(actor) do
+        ActiveRecord::Base.transaction do
+          ApplicationStateChange.new(application_choice).make_offer!
+
+          application_choice.current_course_option = course_option
+          application_choice.offer = { 'conditions' => conditions }
+          application_choice.offered_at = Time.zone.now
+          application_choice.save!
+        end
+
+        SetDeclineByDefault.new(application_form: application_choice.application_form).call
+        SendNewOfferEmailToCandidate.new(application_choice: application_choice).call
       end
+    else
+      raise ValidationException, offer.errors.map(&:message)
     end
   end
+
+private
+
+  def auth
+    @auth ||= ProviderAuthorisation.new(actor: actor)
+  end
+
+  def offer
+    @offer ||= OfferValidations.new(course_option: course_option,
```

Comment:
I'm getting `uninitialized constant MakeOffer::OfferValidations` whilst testing locally, we need to ensure service models is being loaded, or move the validation to the models

Reply:
or maybe services 🤔

Reply:
Autoload paths automatically pick any custom directories under app, so we shouldn't. (https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#autoload-paths)

Did you restart your server?

Reply:
Maybe change it to `::OfferValidations` to ensure that it's not attempting to use local scope?

Reply:
bah yes I restarted and worked! nvm then :D